sdk/core/Azure.Core/tests/TestFramework/RecordedTestBase.cs

```diff
@@ -19,55 +18,58 @@ public abstract class RecordedTestBase: ClientTestBase
 
         protected TestRecording Recording { get; private set; }
 
-        protected RecordedTestBase(bool isAsync) : base(isAsync)
+        protected RecordedTestMode Mode { get; }
+
+        protected RecordedTestBase(bool isAsync) : this(isAsync, GetModeFromEnvironment())
+        {
+        }
+
+        protected RecordedTestBase(bool isAsync, RecordedTestMode mode) : base(isAsync)
         {
             Sanitizer = new RecordedTestSanitizer();
             Matcher = new RecordMatcher(Sanitizer);
+            Mode = mode;
         }
 
-        protected virtual RecordedTestMode Mode
+        private static RecordedTestMode GetModeFromEnvironment()
         {
-            get
+            string modeString = Environment.GetEnvironmentVariable(ModeEnvironmentVariableName);
+
+            RecordedTestMode mode = RecordedTestMode.Playback;
+            if (!string.IsNullOrEmpty(modeString))
             {
-                string modeString = Environment.GetEnvironmentVariable(ModeEnvironmentVariableName);
+                mode = (RecordedTestMode)Enum.Parse(typeof(RecordedTestMode), modeString, true);
```

Comment:
This will fail if the value is misspelled. Should we use `Enum.TryPare` instead and throw a better error message in that case?

Reply:
The exception that is being thrown is not that bad I think it would be pretty obvious where the problem is.